packages/models/proto/v2/delegation/delegation.proto

```diff
@@ -0,0 +1,30 @@
+syntax = "proto3";
+
+package delegation.v2;
+
+message DelegationV2 {
+  string id = 1;
+  string delegatorId = 2;
+  string delegateId = 3;
+  string eserviceId = 4;
+  int64 createdAt = 5;
+  int64 submittedAt = 6;
+  optional int64 approvedAt = 7;
+  optional int64 rejectedAt = 8;
+  optional string rejectionReason = 9;
+  optional int64 revokedAt = 10;
+  DelegationStateV2 state = 11;
+  DelegationKindV2 kind = 12;
+}
+
+enum DelegationStateV2 {
+  WAITING_FOR_APPROVAL= 0;
+  ACTIVE= 1;
+  REJECTED= 2;
+  REVOKED= 3;
+}
+
+enum DelegationKindV2 {
+  DELEGATED_CONSUMER = 1;
+  DELEGATED_PRODUCER = 0;
```

Comment:
@Carminepo2 as discuss in offline conversation with @beetlecrunch, I have renamed Delegation Kind using suggested naming convention Consumer/Producer.